### DISCUSSION TITLE
How to get a dependency from dependency container?

### DISCUSSION THREAD

**Discussion by ghost** [serenity-is/Serenity#6223] [Help] [answered]
Hey, instead of injecting a dependency into constructor or method, I want to get/resolve it straight from the dependency container.

Basically I want to achieve something like this
`var myService = DependencyProvider.Get<IMyService>();`

Thank you!

**Comment by dfaruque** [comment, marked as answer]
You could declare a static property in the Startup class like the following
`public static IServiceProvider ServiceProvider { get; set; }`
then in `Configure` method
`ServiceProvider = app.ApplicationServices;`

Now you can call 
`var myService = Startup.ServiceProvider.GetRequiredService<IMyService>();`

Warning: don't know the side effect of doing this!

**Reply by ghost** [reply]
thank you
